### PR TITLE
[chore] Add contrib confmap providers to otelcontribcol

### PR DESCRIPTION
**Pull request by evan-bradley**
**Description:**

Contrib hosts two confmap providers which aren't in the local binary. We should add them so we can test them with a live system.

cc @Aneurysm9 @driverpt @atoulme

### CODE REVIEW

File: cmd/otelcontribcol/go.mod [hunk 1]
```diff
@@ -4,9 +4,11 @@ module github.com/open-telemetry/opentelemetry-collector-contrib/cmd/otelcontrib
 
 go 1.21.0
 
-toolchain go1.21.9
+toolchain go1.21.5
```
Comment (songy23) [inline cmd/otelcontribcol/go.mod:7]: Is this needed?

Comment (evan-bradley) [reply cmd/otelcontribcol/go.mod]: My bad, definitely not needed. Thanks for catching that, I've reverted this.

**Review by Aneurysm9 — APPROVED** [review]

**Comment by Aneurysm9** [comment]
It looks like something else has a dependency on `github.com/aws/aws-sdk-go-v2/feature/s3/manager@v1.11.59` which may need to be updated to function correctly with the newer version of `github.com/aws/aws-sdk-go-v2/service/s3`.  We have long included the S3 config provider in the ADOT distribution and [do not have](https://github.com/aws-observability/aws-otel-collector/blob/main/go.mod#L125-L126) an implicit dependency on `github.com/aws/aws-sdk-go-v2/feature/s3/manager`.

**Comment by evan-bradley** [comment]
Thank you for the details. I'm digging into it right now, it seems like we don't have a direct dependency on the old version and this error is sourcing from one of our dependencies that pulls in `github.com/aws/aws-sdk-go-v2/feature/s3/manager` either directly or transitively.

**Review by atoulme — APPROVED** [review]
I approve this change and am excited for this new functionality!

**Comment by evan-bradley** [comment]
It looks like the snowflake receiver pulls in an old version of the library: https://github.com/snowflakedb/gosnowflake/blob/master/go.mod#L12. I'm not 100% sure how to resolve this outside a replace statement or making a PR against the library to upgrade the version of the AWS packages they use. Any input is welcome, otherwise I'll dig deeper tomorrow. I suspect this will also impact the awss3receiver, which [pulls in the latest version](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/awss3receiver/go.mod#L8).

**Review by bryan-aguilar — APPROVED** [review]

**Comment by mx-psi** [comment]
I am in favor of trying to fix this upstream first before adding any replaces

**Comment by evan-bradley** [comment]
Fully agreed, I _really_ would prefer to not have any replace statements. I've issued a PR upstream that I have confirmed fixes this when building the binary locally: https://github.com/snowflakedb/gosnowflake/pull/1125.

**Review by mx-psi — APPROVED** [review]

**Comment by evan-bradley** [comment]
I've updated the Snowflake receiver to pull from the gosnowflake commit that updates the AWS SDK dependency. The commit is the first commit after the recent 1.10.0 release of gosnowflake.

All tests pass, so I don't think it should be an issue, but could one of you please take a look as code owners @dmitryax @shalper2?

**Review by shalper2 — APPROVED** [review]
LGTM -- tests are passing for me locally as well.

**Comment by evan-bradley** [comment]
I think we have broad support for this change and for taking on the dependency version pinned on a commit, so I am merging this.

**Pull request merged**